controllers/statusreport/statusreport_adapter.go

```diff
@@ -0,0 +1,104 @@
+	"github.com/redhat-appstudio/integration-service/status"
+
+	"github.com/redhat-appstudio/integration-service/loader"
+	"github.com/redhat-appstudio/operator-toolkit/controller"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Adapter holds the objects needed to reconcile a snapshot's test status report.
+type Adapter struct {
+	snapshot    *applicationapiv1alpha1.Snapshot
+	application *applicationapiv1alpha1.Application
+	logger      helpers.IntegrationLogger
+	loader      loader.ObjectLoader
+	client      client.Client
+	context     context.Context
+	status      status.Status
+}
+
+// NewAdapter creates and returns an Adapter instance.
+func NewAdapter(snapshot *applicationapiv1alpha1.Snapshot, application *applicationapiv1alpha1.Application, logger helpers.IntegrationLogger, loader loader.ObjectLoader, client client.Client,
+	context context.Context) *Adapter {
+	return &Adapter{
+		snapshot:    snapshot,
+		application: application,
+		logger:      logger,
+		loader:      loader,
+		client:      client,
+		context:     context,
+		status:      status.NewAdapter(logger.Logger, client),
+	}
+}
+
+// EnsureSnapshotTestStatusReported will ensure that integration test status including env provision and snapshotEnvironmentBinding error is reported to the git provider
+// which (indirectly) triggered its execution.
+func (a *Adapter) EnsureSnapshotTestStatusReported() (controller.OperationResult, error) {
+	reporters, err := a.status.GetReporters(a.snapshot)
+	if err != nil {
+		return controller.RequeueWithError(err)
+	}
+
+	for _, reporter := range reporters {
+		integrationTestStatusDetails, err := getIntegrationScenarioTestStatusFromAnnotation(a.snapshot)
+		if err != nil {
+			a.logger.Error(err, "failed to get test status to snapshot annotations for snapshot",
+				"snapshot.Namespace", a.snapshot.Namespace, "snapshot.Name", a.snapshot.Name)
+			return controller.RequeueWithError(err)
+		}
+		if integrationTestStatusDetails == nil {
+			a.logger.Info("no snapshot annotation test.appstudio.openshift.io/status defined for snapshot, no need to report integration test status",
+				"snapshot.Namespace", a.snapshot.Namespace, "snapshot.Name", a.snapshot.Name)
+			return controller.ContinueProcessing()
+		}
+
+		if err := reporter.ReportStatusForSnapshot(a.client, a.context, &a.logger, a.snapshot, integrationTestStatusDetails); err != nil {
+			a.logger.Error(err, "failed to report test status to github for snapshot",
+				"snapshot.Namespace", a.snapshot.Namespace, "snapshot.Name", a.snapshot.Name)
+			return controller.RequeueWithError(err)
+		}
+	}
+
+	return controller.ContinueProcessing()
+}
+
+// getIntegrationScenarioTestStatusFromAnnotation get the test status list for all integration test scenarios of snapshot
+// then Unmarshal to get structured data, return error if Unmarshal fails
+func getIntegrationScenarioTestStatusFromAnnotation(snapshot *applicationapiv1alpha1.Snapshot) (*[]gitops.IntegrationTestStatusDetail, error) {
```

Comment:
Please use `gitops.NewSnapshotIntegrationTestStatusesFromSnapshot(snapshot).GetStatuses()`

This is just duplicate of that function

Reply:
Updated. Thanks